custom_components/myskoda/coordinator.py

```diff
@@ -271,16 +271,19 @@ async def _on_mqtt_event(self, event: Event) -> None:
             self.service_events.appendleft(event.event)
             self.async_set_updated_data(self.data)
 
-            if event.topic == ServiceEventTopic.CHARGING:
-                await self._on_charging_event(event)
-            if event.topic == ServiceEventTopic.ACCESS:
-                await self._on_access_event(event)
-            if event.topic == ServiceEventTopic.AIR_CONDITIONING:
-                await self._on_air_conditioning_event(event)
-            if event.topic == ServiceEventTopic.DEPARTURE:
-                await self._on_departure_event(event)
-            if event.topic == ServiceEventTopic.ODOMETER:
-                await self._on_odometer_event(event)
+            service_event_handlers: Dict[
+                ServiceEventTopic, Callable[[Any], Awaitable[None]]
+            ] = {
+                ServiceEventTopic.CHARGING: self._on_charging_event,
+                ServiceEventTopic.ACCESS: self._on_access_event,
+                ServiceEventTopic.AIR_CONDITIONING: self._on_air_conditioning_event,
+                ServiceEventTopic.DEPARTURE: self._on_departure_event,
+                ServiceEventTopic.ODOMETER: self._on_odometer_event,
+            }
+
+            handler = service_event_handlers.get(event.topic)
+            if handler:
+                await handler(cast(Any, event))
```

Comment:
```suggestion
            if handler := service_event_handlers.get(event.topic):
                await handler(cast(Any, event))
```

Or, I think it might make sense to not be silent if somehow we get a eventopic which isn't in the mapping:

```suggestion
            try:
                await service_event_handlers[event.topic](cast(Any, event))
            except KeyError:
                _LOGGER.warning(f"Unhandled event topic {event.topic}")
```

Also don't love the use of `Any` and even less so having to cast to it 😅 . The lose useful type checking doing it this way. Previously if you'd accidentally try to call `self._on_charging_event(event)` with a access event instead of a charging event you'd get a pyright/pylance warning. I'm sure there is a better way to do this than what we currently have.

Reply:
Yeah, I understand what you mean, but I haven't found it yet :slightly_smiling_face: